strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/storage/metadata/nuget/TempNupkgFileTest.java

```diff
@@ -45,97 +43,67 @@
-        throws IOException,
-               JAXBException
-    {
-        getCleanBaseDirectory();
-    }
-
-    private File getCleanBaseDirectory()
-        throws IOException
-    {
-        File baseDirectory = new File(baseDirectoryPath);
-
-        if (baseDirectory.exists())
-        {
-            FileUtils.deleteDirectory(baseDirectory);
-        }
-
-        return baseDirectory;
-    }
+    private static final String REPOSITORY_RELEASES = "tnft-releases";
 
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class,
+                  ArtifactManagementTestExecutionListener.class })
     @Test
-    public void testHashTempFile()
-        throws Exception
+    public void testHashTempFile(@NugetRepository(repositoryId = REPOSITORY_RELEASES)
+                                 Repository repository,
+                                 @NugetTestArtifact(repositoryId = REPOSITORY_RELEASES,
+                                                    id = "NUnit",
+                                                    versions = "2.5.9.10348")
+                                 Path artifactNupkgPath)
+            throws Exception
     {
         // GIVEN
-        String packageId = "NUnit";
-        String packageVersion = "2.5.9.10348";
-        Path packageFilePath = TestCaseWithNugetArtifactGeneration.generateArtifactFile(baseDirectoryPath,
-                                                                                        packageId,
-                                                                                        packageVersion);
-
-        String checksumFileName = packageId + "." + packageVersion + ".nupkg.sha512";
-        Path checksumPath = packageFilePath.resolveSibling(checksumFileName);
+        Path checksumPath = Paths.get(artifactNupkgPath + ".sha512");
 
         String expectedHash = MessageDigestUtils.readChecksumFile(checksumPath.toString());
 
         // WHEN
-        try (InputStream nupkgInputStream = new BufferedInputStream(Files.newInputStream(packageFilePath));
-                TempNupkgFile nupkgFile = new TempNupkgFile(nupkgInputStream);)
+        try (InputStream nupkgInputStream = new BufferedInputStream(Files.newInputStream(artifactNupkgPath));
+             TempNupkgFile nupkgFile = new TempNupkgFile(nupkgInputStream))
         {
             // THEN
-            assertNotNull(nupkgFile.getHash().toString(),
+            assertNotNull(nupkgFile.getHash(),
                           "Hash file created from stream");
             assertEquals(expectedHash,
-                         nupkgFile.getHash().toString(),
+                         nupkgFile.getHash(),
                          "Hash file created from stream");
         }
     }
 
     /**
      * Check reading specifications
      *
-     * @throws Exception
-     *             error during the test
+     * @throws Exception error during the test
      */
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class,
```

Comment:
same as above, and we don't need `RepositoryManagementTestExecutionListener` as well.